saleor/graphql/translations/mutations/sale_translate.py

```diff
@@ -34,8 +44,49 @@ class Meta:
     def perform_mutation(  # type: ignore[override]
         cls, root, info: ResolveInfo, /, *, id, input, language_code
     ):
-        response = super().perform_mutation(
-            root, info, id=id, input=input, language_code=language_code
+        instance = cls.get_promotion_instance(id)
+        cls.validate_input(input)
+
+        input = cls.pre_update_or_create(instance, input)
+
+        translation, created = instance.translations.update_or_create(
+            language_code=language_code, defaults=input
         )
-        instance = ChannelContext(node=response.sale, channel_slug=None)
-        return cls(**{cls._meta.return_field_name: instance})
+        manager = get_plugin_manager_promise(info.context).get()
+
+        if created:
+            cls.call_event(manager.translation_created, translation)
+        else:
+            cls.call_event(manager.translation_updated, translation)
+
+        return cls(
+            **{
+                cls._meta.return_field_name: ChannelContext(
+                    node=instance, channel_slug=None
+                )
+            }
+        )
+
+    @classmethod
+    def get_promotion_instance(cls, id):
+        if not id:
+            raise ValidationError(
+                {
+                    "id": ValidationError(
+                        "This field is required", code=DiscountErrorCode.REQUIRED.value
+                    )
+                }
+            )
+        type, node_pk = from_global_id_or_error(id, raise_error=False)
+        if type == "Promotion" or type == "PromotionTranslatableContent":
```

Comment:
Could it be just
```suggestion
        if type in ("Promotion", "PromotionTranslatableContent"):
```
?